### PR TITLE
Refactor Tablet for cloud object storage

### PR DESCRIPTION
**Pull request by wyb**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Problem Summary(Required) ：

The original Tablet is abstracted into two subclasses.
1. LocalTablet is based on local disk storage and replicas are managed by StarRocks.
2. StarOSTablet is based on cloud object storage. Data replicas are managed by object storage
and compute replicas are managed by StarOS through Shard.

Update the Gson persistence framework.
1. Support deserializing LocalTablet from old Tablet serialized data when upgrading.
2. Support deserializing old Tablet from LocalTablet serialized data when rolling back.

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/catalog/LocalTablet.java [hunk 1]
```diff
@@ -0,0 +1,609 @@
...
+                if (!tablet.replicas.contains(replicas.get(i))) {
+                    return false;
+                }
+            }
+        }
+        return id == tablet.id;
+    }
+
+    @Override
+    public long getDataSize() {
```
Comment (wuleistarrocks) [inline fe/fe-core/src/main/java/com/starrocks/catalog/LocalTablet.java:383]: is it logical right to return the total size of each replica? and it seems a little different from the original Tablet::getDataSize(boolean singleReplica), may be you should check this carefully

Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/catalog/LocalTablet.java:383]: There is only one place `MaterializedIndex.getDataSize()` where this parameter is used, and it is false.

**Comment by wyb** [comment]
run starrocks_fe_unittest

**Review by gengjun-git — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/com/starrocks/catalog/LocalTablet.java (hunk 1 above)
Comment (wuleistarrocks) [reply fe/fe-core/src/main/java/com/starrocks/catalog/LocalTablet.java:383]: LGTM

**Review by sduzh — APPROVED** [review]

**Pull request merged**